### PR TITLE
Pull Radarr MediaInfo Updates/Fixes

### PR DESCRIPTION
**Pull request by geogolem**
#### Database Migration
YES

#### Description
This pulls Radarr changes: 
https://github.com/Radarr/Radarr/pull/5352 and https://github.com/Radarr/Radarr/issues/5404

#### Todos
- [ ] Tests
- [ ] Documentation

#### Issues Fixed or Closed by this PR

*

### CODE REVIEW

File: src/NzbDrone.Core.Test/Datastore/Migration/188_mediainfo_channel_propertiesFixture.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+{
+    [TestFixture]
+    public class mediainfo_channel_propertiesFixture : MigrationTest<mediainfo_channels>
+    {
+        private void AddMovieFile(mediainfo_channels m, int id)
+        {
+            var movie = new
+            {
+                Id = id,
+                MovieId = id,
```
Comment (geogolem) [inline src/NzbDrone.Core.Test/Datastore/Migration/188_mediainfo_channel_propertiesFixture.cs:19]: I still need to fix this...

Comment (geogolem) [reply src/NzbDrone.Core.Test/Datastore/Migration/188_mediainfo_channel_propertiesFixture.cs:19]: Also I probably need to change the migration number

File: src/NzbDrone.Core.Test/Datastore/Migration/188_mediainfo_channel_propertiesFixture.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                Quality = new { }.ToJson(),
+                Size = 0,
+                DateAdded = DateTime.UtcNow,
+                RelativePath = "SomeFile.mkv",
+                Languages = new[] { 1, 2 }.ToJson(),
```
Comment (geogolem) [inline src/NzbDrone.Core.Test/Datastore/Migration/188_mediainfo_channel_propertiesFixture.cs]: Im pretty sure Languages and a few other properties here need to be changed for Sonarr..

File: src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        {
+            var episode = new
+            {
+                Id = id,
+                SeriesId = id,
+                Quality = new { }.ToJson(),
+                Size = 0,
+                DateAdded = DateTime.UtcNow,
+                RelativePath = "SomeFile.mkv",
+                Languages = new[] { 1, 2 }.ToJson(),
```
Comment (geogolem) [inline src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs:24]: Im pretty sure Languages and a few other properties here need to be changed for Sonarr..

Comment (Qstick) [reply src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs:24]: Yea there are a few fixups on these to make them Sonarr compatible

File: src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    [Migration(148)]
+    public class mediainfo_channels : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Execute.Sql("UPDATE EpisodeFiles SET MediaInfo = Replace(MediaInfo, '\"audioChannels\"', '\"audioChannelsContainer\"');");
+            Execute.Sql("UPDATE EpisodeFiles SET MediaInfo = Replace(MediaInfo, '\"audioChannelPositionsText\"', '\"audioChannelPositionsTextContainer\"');");
+
+            //Change List Interval from Min to Hour
+            Execute.Sql("UPDATE Config SET Value = max((Value / 60) + 1, 6) WHERE Key = 'importlistsyncinterval'");
```
Comment (Qstick) [inline src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs:15]: Get rid of this for Sonarr

File: src/NzbDrone.Core/MediaFiles/MediaInfo/VideoFileInfoReader.cs [hunk 5]
```diff
@@ -18,8 +18,8 @@ public class VideoFileInfoReader : IVideoFileInfoReader
         private readonly IDiskProvider _diskProvider;
         private readonly Logger _logger;
 
-        public const int MINIMUM_MEDIA_INFO_SCHEMA_REVISION = 3;
-        public const int CURRENT_MEDIA_INFO_SCHEMA_REVISION = 5;
+        public const int MINIMUM_MEDIA_INFO_SCHEMA_REVISION = 4;
```
Comment (Qstick) [inline src/NzbDrone.Core/MediaFiles/MediaInfo/VideoFileInfoReader.cs:21]: Keep the minimum for Sonarr at 3, Radarr minimum was bumped on v0.2 in a previous change

File: src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                IndexerFlags = 0,
```
Comment (Qstick) [inline src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs:25]: No indexer flags in Sonarr

**Review by Qstick — CHANGES_REQUESTED** [review]

File: src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs (hunk 4 above)
Comment (geogolem) [reply src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs:15]: You mean the entire file, right?

Comment (Qstick) [reply src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs:15]: Yep

File: src/NzbDrone.Core/MediaFiles/MediaInfo/MediaInfoFormatter.cs [hunk 7]
```diff
@@ -468,41 +469,71 @@ public static string FormatVideoCodecLegacy(MediaInfoModel mediaInfo, string sce
...
         {
-            var audioChannelPositionsText = mediaInfo.AudioChannelPositionsText;
-            var audioChannels = mediaInfo.AudioChannels;
+            var audioChannelPositionsTextContainer = mediaInfo.AudioChannelPositionsTextContainer;
+            var audioChannelPositionsTextStream = mediaInfo.AudioChannelPositionsTextStream;
+            var audioChannelsContainer = mediaInfo.AudioChannelsContainer;
+            var audioChannelsStream = mediaInfo.AudioChannelsStream;
 
-            if (audioChannelPositionsText.IsNullOrWhiteSpace())
+            //Skip if the positions texts give us nothing
```
Comment (markus101) [inline src/NzbDrone.Core/MediaFiles/MediaInfo/MediaInfoFormatter.cs:488]: Nit: space between `//` and comment text

File: src/NzbDrone.Core/MediaFiles/MediaInfo/VideoFileInfoReader.cs [hunk 8]
```diff
@@ -141,8 +142,12 @@ public MediaInfoModel GetMediaInfo(string filename)
                     string audioChannelsStr = mediaInfo.Get(StreamKind.Audio, 0, "Channel(s)").Split(new string[] { " /" }, StringSplitOptions.None)[0].Trim();
                     int.TryParse(audioChannelsStr, out audioChannels);
 
-                    var audioChannelPositions = mediaInfo.Get(StreamKind.Audio, 0, "ChannelPositions/String2");
+                    string audioChannelsStrOrig = mediaInfo.Get(StreamKind.Audio, 0, "Channel(s)_Original").Split(new string[] { " /" }, StringSplitOptions.None)[0].Trim();
```
Comment (markus101) [inline src/NzbDrone.Core/MediaFiles/MediaInfo/VideoFileInfoReader.cs:145]: Nit: `var` instead of `string` (and `audioChannelsStr` above).

**Review by markus101 — CHANGES_REQUESTED** [review]
Just minor things then looks good.

File: src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs [hunk 9]
```diff
@@ -0,0 +1,104 @@
+using System;
```
[the remark this answers is not in the document]
Comment (geogolem) [reply src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs:1]: I think it does, does it not? I might be missing something here, but isn't it 133?

[the remark this answers is not in the document]
Comment (markus101) [reply src/NzbDrone.Core.Test/Datastore/Migration/133_mediainfo_channel_propertiesFixture.cs:1]: The migration is numbered 148.

**Review by markus101 — CHANGES_REQUESTED** [review]
Please squash the fixups into the original commits and we'll get this landed.

File: src/NzbDrone.Core.Test/Datastore/Migration/148_mediainfo_channel_propertiesFixture.cs [hunk 10]
```diff
@@ -11,18 +11,17 @@ namespace NzbDrone.Core.Test.Datastore.Migration
     [TestFixture]
     public class mediainfo_channel_propertiesFixture : MigrationTest<mediainfo_channels>
     {
-        private void AddMovieFile(mediainfo_channels m, int id)
+        private void AddEpisodeFile(mediainfo_channels m, int id)
```
Comment (markus101) [inline src/NzbDrone.Core.Test/Datastore/Migration/148_mediainfo_channel_propertiesFixture.cs:14]: This was squashed into the wrong commit.

File: src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs [hunk 11]
```diff
@@ -0,0 +1,15 @@
+using FluentMigrator;
```
Comment (markus101) [inline src/NzbDrone.Core/Datastore/Migration/148_mediainfo_channels.cs:1]: This one too.

File: src/NzbDrone.Core/MediaFiles/MediaInfo/VideoFileInfoReader.cs [hunk 12]
```diff
@@ -18,7 +18,7 @@ public class VideoFileInfoReader : IVideoFileInfoReader
         private readonly IDiskProvider _diskProvider;
         private readonly Logger _logger;
 
-        public const int MINIMUM_MEDIA_INFO_SCHEMA_REVISION = 4;
+        public const int MINIMUM_MEDIA_INFO_SCHEMA_REVISION = 3;
```
Comment (markus101) [inline src/NzbDrone.Core/MediaFiles/MediaInfo/VideoFileInfoReader.cs:21]: This should be in the previous commit as well.

**Review by markus101 — CHANGES_REQUESTED** [review]
Looks like things were squashed into the wrong commit, which makes the commit history messier than it needs to be.

**Comment by geogolem** [comment]
@markus101 : I had to do some git acrobatics to get those squashes into the right commits --> I didnt really know how to do it, but had to learn/figure it out.

I think its all OK now.. but please make sure.. 

Thanks

**Pull request merged**